server/src/main/java/org/opensearch/search/aggregations/bucket/terms/GlobalOrdinalsStringTermsAggregator.java

```diff
@@ -603,6 +603,15 @@ abstract class ResultStrategy<
         TB extends InternalMultiBucketAggregation.InternalBucket> implements Releasable {
 
         private InternalAggregation[] buildAggregations(long[] owningBucketOrds) throws IOException {
+            int requiredSizeLocal;
+            long minDocCountLocal;
+            if (context.isConcurrentSegmentSearchEnabled()) {
```

Comment:
The `bucketCountThresholds` here comes from `TermsAggregator`, which I have not modified, as opposed to coming from `InternalTerms`. For concurrent search we don't want to apply any thresholds at the slice level and let this happen during `reduce` instead. In general I don't really like modifying the values of `bucketCountThresholds` since it already contains both shard and coordinator level thresholds so it seems less confusing to me to pick one based on a flag.

Reply:
> What I am suggesting is to have a methods in reduceContext and context to return LocalBucketCountThresholds based a) isSliceLevel b) isConcurrentSegmentSearchEnabled. The would be a single place to make the decision.

Ah okay I misunderstood what you were suggesting at first. This is a good idea I will make this change.

Reply:
Added this to context and reduceContext in the latest revision